woocommerce/google-listings-and-ads
# Product Block Editor: Convert `Label` component and custom block registration to use the newly added APIs in Woo 8.5

### Changes proposed in this Pull Request:

To be compatible with Product Block Editor and to use Woo core APIs as much as possible, this PR converts a few codes to rely on Woo 8.5:

- Use Woo's product block template API to register custom blocks. (Ref: [woocommerce#41937](https://github.com/woocommerce/woocommerce/pull/41937))
- Replace the copied `Label` component with the one imported from `@woocommerce/product-editor`. (Ref: [woocommerce#42959](https://github.com/woocommerce/woocommerce/pull/42959))

### Screenshots:

#### 📷 Label of the Product select with text field block

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/fb007c75-cc92-4232-9e9b-986269a2951e)

#### 📷 Label of the Product select field block

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/8d5db0cb-f69e-4cb0-bb3f-7c3f0659791d)

### Detailed test instructions:

#### 📌 Prepare test environment

1. Please refer to [the test preparation in PR 2151](https://github.com/woocommerce/google-listings-and-ads/pull/2151#prepare-test-environment) with **Woo 8.5.0**.
   - Currently, it can be tested with [8.5.0-rc.1](https://github.com/woocommerce/woocommerce/releases/tag/8.5.0-rc.1)
1. `npm install`
1. `npm start`

#### 📌 Test the changes

1. Install [WooCommerce Brands](https://github.com/woocommerce/woocommerce-brands) and activate it.
1. Go to edit a variable product, e.g. the **Hoodie** imported from WC's sample products.
1. Switch to the **Google Listings & Ads** tab.
1. Check if there are only two attributes in the **Product attributes** sections and if the attributes can access their product data.
   - There are features specific to the product block template and they require specific block attributes such as `_templateBlockHideConditions` and `usesContext` to make them work.
   - The `_templateBlockHideConditions` is for conditional hiding a block.
   - The use of `usesContext` is explained [here](https://github.com/woocommerce/google-listings-and-ads/blob/2d84378cf93848f1eeab046973e2d1ce127fd3ae/js/src/blocks/README.md#compatible-with-block-templates-and-product-data-access).
   - After removing _templateBlockHideConditions from block.json files, attributes that are not applicable for variable product should still be hidden, e.g. [Color](https://github.com/woocommerce/google-listings-and-ads/blob/2d84378cf93848f1eeab046973e2d1ce127fd3ae/src/Product/Attributes/Color.php#L38-L40).
   - After removing the `usesContext` from block.json files, the accessing of attributes should still work well.
1. Hover on the question icon of **Brand** (Product select with text field block) to see if the label content shows up.
1. Hover on the question icon of **Adult content** (Product select field block) to see if the label content shows up.

### Changelog entry
